frontend/src/store/index.ts

```diff
@@ -70,19 +76,20 @@ export default createStore({
       });
       context.commit('login', { id: context.getters.id, token: response.data });
     },
+    async getAvatarById(context, id: string) {
+      if (id) {
+        if (context.state.cache?.avatars.has(+id)) {
+          return context.state.cache?.avatars.get(+id);
+        }
+        return await fetchAvatar(+id);
+      }
+    },
     async getAvatar(context) {
-      try {
-        const id = context.state.user.id;
-        if (id === undefined) return undefined;
-        const response = await axios.get('/users/' + id + '/avatar', {
-          responseType: 'blob',
-        });
-        if (response.status === 204) return;
-        const blob = new Blob([response.data]);
-        context.state.avatar = URL.createObjectURL(blob);
-      } finally {
-        if (!context.state.avatar) {
-          context.state.avatar = kingPongImg;
+      if (context?.state?.user?.id) {
+        if (context.state.cache?.avatars.has(+context?.state?.user?.id)) {
+          context.state.avatar = context.state.cache?.avatars.get(
+            +context?.state?.user?.id,
+          );
```

Comment:
Ah, sorry, something must have broken when I merged the latest changes from master. Thanks, I'll take a look!

Reply:
Okay, it seems to be working now. The path issue came from trying to delete an avatar file that didn't exist, so I took the fix that is in place on your branch so that it can be tested